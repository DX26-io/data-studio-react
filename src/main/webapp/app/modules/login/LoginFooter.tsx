import React from 'react';
import { Text } from '@adobe/react-spectrum'

import { CommonConstants } from '../../shared/util/common.constants';

// TODO: Test Cases
// TODO : year should come dynamically 

const LoginFooter = props => (
    <Text position="absolute" right="3%" bottom="4%">{CommonConstants.DX26} {CommonConstants.COPYRIGHT} 2020</Text>
);

export default LoginFooter;
