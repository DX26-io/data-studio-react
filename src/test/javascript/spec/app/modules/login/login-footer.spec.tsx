import React from 'react';
import { shallow } from 'enzyme';
import { Text } from '@adobe/react-spectrum';
import LoginFooter from 'app/modules/login/login-footer';

describe('Login Footer', () => {
  let mountedWrapper;

  const wrapper = () => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<LoginFooter />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  it('Renders Login Footer', () => {
    const component = wrapper();
    expect(component).toMatchSnapshot();
    const text = component.find(Text);
    expect(text.length).toEqual(1);
  });
});
