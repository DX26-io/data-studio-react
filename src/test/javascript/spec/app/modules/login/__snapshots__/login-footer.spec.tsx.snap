// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Footer Renders Login Footer 1`] = `
<ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
  direction="row-reverse"
>
  <ForwardRef($fe1955053990ab873941970c7463$var$Text)
    marginEnd="size-300"
  >
    Dx26
     
    Â©
     
    2020
  </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
</ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
`;
